zigpy/device.py

```diff
@@ -41,11 +41,7 @@ def __init__(self, application, ieee, nwk):
         self._node_handle = None
 
     def schedule_initialize(self):
-        if self.initializing:
-            LOGGER.debug("Canceling old initialize call")
-            self._init_handle.cancel()
-        else:
-            self.initializing = True
+        self.initializing = True
```

Comment:
No, if we do `asyncio.ensure_future()` then there's a chance this would be called twice in succession. We do need to cancel old task. 
And please run this change in your production for a week and report back :)